controllers/pvc_storageclass_test.go

```diff
@@ -0,0 +1,320 @@
+package controllers
+
+import (
+	"context"
+	"time"
+
+	"github.com/darkowlzz/operator-toolkit/webhook/cert"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	admissionv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	pvcmutator "github.com/storageos/api-manager/controllers/pvc-mutator"
+	"github.com/storageos/api-manager/controllers/pvc-mutator/storageclass"
+	"github.com/storageos/api-manager/internal/pkg/provisioner"
+)
+
+// Define utility constants for object names and testing timeouts and intervals.
+const (
+	storageClassTimeout  = time.Second * 1
+	storageClassInterval = time.Millisecond * 250
+)
+
+var (
+	defaultStorageClassName        = "stos-default"
+	givenStorageClassName          = "stos"
```

Comment:
Similar to the default SC name, this can be renamed to `stosStorageClassName` for clarity.